homeassistant/components/soma/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Config flow for Soma."""
+import logging
+
+import voluptuous as vol
+from api.soma_api import SomaApi
+from requests import RequestException
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 3000
+
+
+class SomaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if user_input is None:
+            data = {
+                vol.Required(CONF_HOST): str,
+                vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+            }
+
+            return self.async_show_form(step_id="user", data_schema=vol.Schema(data))
+
+        return await self.async_step_creation(user_input)
+
+    async def async_step_creation(self, user_input=None):
+        """Finish config flow."""
+        api = SomaApi(user_input["host"], user_input["port"])
+        try:
+            await self.hass.async_add_executor_job(api.list_devices)
+            _LOGGER.info("Successfully set up Soma Connect")
+            return self.async_create_entry(
+                title="Soma Connect",
+                data={"host": user_input["host"], "port": user_input["port"]},
+            )
+        except RequestException:
```

Comment:
We need to test this case too in the tests:
https://codecov.io/gh/home-assistant/home-assistant/compare/930dadb7226c111658d691c13f2a46d9019d9ea7...df0965d89f83ae7e7511afaf4a856dca0607e8bc/diff

Patch `SomaApi.list_devices` and set a side_effect to `RequestException`. That will raise that error in the test covering that case.